include/deal.II/base/function_lib.h

```diff
@@ -965,6 +1016,92 @@ namespace Functions
      * in all components.
      */
     const unsigned int selected;
+
+    /**
+     * Flag that controls wether we rescale the value when the radius changes.
+     */
+    bool integrate_to_one;
+
+    /**
+     * The reference integral value. Derived classes should specify what their
+     * integral is when @p radius = 1.0.
+     */
+    const double unitary_integral_value;
```

Comment:
Ah, I see how you think of this. You are assuming that the cutoff-function scales somehow. This is a semantic requirement on the derived classes that I think you should document. In particular, it assumes that if the radius of a cutoff function is changed, that the height does not. That seems like a rather severe restriction on derived classes.